### DISCUSSION TITLE
2 One-to-Many or Many-to-Many Relationship

### DISCUSSION THREAD

**Discussion by bernhard-teacher** [Budibase/budibase#9991] [Help] [answered]
Hi all,
i have two tables with m-n Relation and a Through-Table in external PostgreSQL-Database.

```
course         |  z_course_participant | participant
course_id (PK) | course_id (FK)        | participant_id (PK)
               | participant_id (FK)   |
```

The PK of z_course_participant is joined of the FKs.
In the z_ Table there are other Attributes like a partipant-status and so on.
So i need to edit the rows of z_course_participant.

If i define in BudiBase a Many-to-Many-Relationship, i only see the IDs in the FKs, but not the Lookup of the Displaycolumn in the referenced Tables, what is annoying.
But i cannot define 2 one-to-Many-Relationships, the FKs arent displayed in the list and i cannot take the course_id as reference.

So, is it possible to lookup the reference (in a many-to-many-Relation) or to take the FKs as reference for the one-to-many-Relation or do i need an extra PK in the z_Table (which i don't prefer, because for other Design-Structures).

Thanks for your suggestions!

**Comment by melohagan** [comment, marked as answer]
Hey @bernhard-teacher 

I believe some of your issues may be arising from the fact that your joining table contains additional information that is not purely for joining.

> In the z_ Table there are other Attributes like a partipant-status and so on.

The Participant status should be in the Participant table. 

If you are doing a Many->Many relationship it is the correct approach to use a joining table.

**Reply by bernhard-teacher** [reply]
Hi @melohagan ,
thanks for your reply. No, the Participant Status is based on the Joining-Table: there are the registrations and the Status is e.g. "in course" or "on waiting" and so on. So the additional information must be in the joining table.
I resolved it now by tow 1-to-many-Relationships and an extra ID in the joining table...and set the FKs on "not NULL". Perhaps not the best solution, but it works now for me ;)

Thanks for your support!

**Reply by melohagan** [reply]
Where the FKs actually defined with Foreign Key constraints in the SQL schema? 

If not, then what you've got is grand!

**Reply by bernhard-teacher** [reply]
yes, they were defined as FK constraints and together as PK. Now i builded a new extra PK and the FK constraints rebuilded - and with them defined the 1-to-many-Relationship now...
